### DISCUSSION TITLE
Thoughts on the governance of cFS as an open-source software

### DISCUSSION THREAD

**Discussion by ugo94490** [nasa/cFS#309] [Ideas]
Hi everyone,

Recently we had several ideas for improvements with @ThibFrgsGmz and we wanted to post them here to get some feedback from the community.

# Product Line

We personally think that cFS would benefit from being managed by an open-source foundation to be more open to the OpenSource/International community.

As a project successfully managed by an Open Source organization we can mention Rust managed by the Mozilla foundation.

This would make it easier to get funding from governmental institutions, universities, or private sector companies to develop Open Source on this project.

We could imagine that big companies like GAFA wanting to contribute to the development of space, sponsor the foundation, like Microsoft / AWS / Google / Huawei / Mozilla for the [Rust language with the Mozilla Foundation](https://research.mozilla.org/rust/).

Exporting part of the development and maintenance to an external/independent organization could perhaps also reassure companies worried about the long-term support of cFS.

This foundation would be responsible for continuing to develop/maintain cFS for the entire community in the same way that NASA does.

Each contributor will still be able to continue to spontaneously/ponically improve cFS when implementing it in personal projects.

If this option is not possible in the short or medium term, one can imagine cFS into a community-centered project.
For example like [F Prime](https://github.com/nasa/fprime/) (JPL), [VSCode](https://github.com/microsoft/vscode) (Microsoft), [React](https://github.com/facebook/react) (Facebook), [BootStrap](https://github.com/twbs/bootstrap) which were created by companies but are still maintained by the community.

We have the feeling that cFS tends to open more and more to the community with for example the implementation of GitHub discussions last month. 

Nevertheless, we should make the community feel more involved in the improvement of cFS by adding, for example, a "Looking For Contributor" tag on issues that NASA would not have the time to integrate but that could be useful to cFS.

We could also imagine filling the [Wiki](https://github.com/nasa/cFS/wiki) tab with tutorials and documentation, using more the [Projects](https://github.com/nasa/cFS/projects) tab to show graphically to the community what are the upcoming improvements and issues to fix and the status of these tasks.

<br/>
<br/>
We would be delighted to hear your opinions on these various improvements and to discuss with you in order to improve them or to add others.

**Comment by skliper** [comment]
We've also discussed a quarterly (give or take) community tag-up to increase awareness and hopefully collaboration across the community.  Basically covering updates, current work, future opportunities, coordination, etc.

**Reply by jbohren-hbr** [reply]
We'd be interested in such a quarterly meeting.

**Comment by thnkslprpt** [comment]
The foundation idea sounds nice in principle, but I would be worried it would introduce a lot of bureaucracy and politics...

Improving the Wiki would be awesome. Simple video tutorials (basic build+run type stuff) would be cool - perhaps someone has experience with recording these type of videos?

**Reply by jphickey** [reply]
> would introduce a lot of bureaucracy and politics

More than the US government?? .... :-)

**Reply by thnkslprpt** [reply]
haha - I was more worried that the bureaucracy and politics would be combined/added, rather than replace...

**Comment by jbohren-hbr** [comment]
@ugo94490 From my read of your post it sounds like you've identified a few issues that you would intend a 3rd-party foundation would address:

1. Increase funding for cFS development,
2. Reduce perceived risk in cFS adoption,
3. Increase community involvement for cFS, and
4. Increase accessibility through documentation

1 is something that having a foundation won't necessarily solve, unless that foundation intends to provide cFS consulting and engineering services which lead to cFS contributions. Then it's less of a foundation and really just an engineering services company.

2, similarly to 1, will really only happen if an organization exists which can provide a cFS long-term support contract with some legal assurance. 

Regarding 3, I've been really impressed with the attention that the cFS stakeholders at NASA have given to the community, and it feels like it's moving in the right direction. Even our most recent (albeit trivial) pull request was given a solid amount of attention and critical feedback, documented entirely in the open.

In terms of 4, it's important to identify stakeholders to whom you're trying to make cFS more accessible. This could lead to very different products depending on the target. Also more documentation is great, but it also increases the maintenance burden as the software evolves.
